### PR TITLE
Update return to 500 following config update error

### PR DESCRIPTION
**Pull request by cdjackson**
getThingNotFoundResponse doesn't seem an appropriate response since we already checked that the thing is found a few lines earlier. I had an error in a handler that gave thing not found and it makes it hard to trace.

I suggest to change to 500 "Internal Server Error".

Signed-off-by: Chris Jackson <chris@cd-jackson.com>

### CODE REVIEW

File: bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/thing/ThingResource.java [hunk 1]
```diff
@@ -493,7 +494,7 @@ public Response updateConfiguration(@HeaderParam(HttpHeaders.ACCEPT_LANGUAGE) St
         } catch (IllegalArgumentException ex) {
             logger.info("Received HTTP PUT request for update config at '{}' for the unknown thing '{}'.",
                     uriInfo.getPath(), thingUID);
-            return getThingNotFoundResponse(thingUID);
+            return Response.status(Status.INTERNAL_SERVER_ERROR).build();
```
Comment (tomhoefer) [inline bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/thing/ThingResource.java]: I think the logged message should be updated as well

**Comment by cdjackson** [comment]
Good point - updated...

**Comment by kaikreuzer** [comment]
Hm, a server 500 error should never be an expected return code - it should only happen, if there is a real bug on the server side and in this case it should be reported as ERROR in the log as well.

Do I understand it correctly that this situation ONLY occurs if there is a real bug in an implementation? Or could it also occur in situations where wrong parameters are passed in the REST call?

**Comment by cdjackson** [comment]
It's catching an exception from the handler - currently only IllegalArgumentException but I did wonder if we shouldn't catch other exceptions here and report the error back to the user?

**Comment by kaikreuzer** [comment]
Yes, we should probably catch Exception in general.
I am still wondering though if we should document the 500 in Swagger - after all, this applies to ALL bugs in REST endpoints, they will always result in a 500; but this is not really a part of the API.

**Comment by cdjackson** [comment]
Ok - fair point. So how about I change the exception to catch all and remove the swagger doc?

**Comment by kaikreuzer** [comment]
Sounds good to me.

**Comment by cdjackson** [comment]
Updated as discussed.

File: bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/thing/ThingResource.java [hunk 2]
```diff
@@ -490,10 +490,10 @@ public Response updateConfiguration(@HeaderParam(HttpHeaders.ACCEPT_LANGUAGE) St
             logger.debug("Config description validation exception occurred for thingUID {} - Messages: {}", thingUID,
                     ex.getValidationMessages());
             return Response.status(Status.BAD_REQUEST).entity(ex.getValidationMessages(locale)).build();
-        } catch (IllegalArgumentException ex) {
-            logger.info("Received HTTP PUT request for update config at '{}' for the unknown thing '{}'.",
-                    uriInfo.getPath(), thingUID);
-            return getThingNotFoundResponse(thingUID);
+        } catch (Exception ex) {
+            logger.info("Exception during HTTP PUT request for update config at '{}' for thing '{}': {}",
```
Comment (kaikreuzer) [inline bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/thing/ThingResource.java]: As suggested, better use error logging to clearly indicate that we stumbled across an implementation bug.

File: bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/thing/ThingResource.java [hunk 3]
```diff
@@ -490,10 +490,10 @@ public Response updateConfiguration(@HeaderParam(HttpHeaders.ACCEPT_LANGUAGE) St
...
+                    uriInfo.getPath(), thingUID, ex.getMessage());
+            return Response.status(Status.INTERNAL_SERVER_ERROR).build();
```
Comment (kaikreuzer) [inline bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/thing/ThingResource.java]: Might it be worthwhile to return a JSONResponse that also passes the exception message to the client?

**Comment by cdjackson** [comment]
Ok, I'll fix these two issues after the other PR is merged as there will be merge conflicts that will need resolving.

**Comment by kaikreuzer** [comment]
"The other" is merged :-)

**Comment by cdjackson** [comment]
Updated...

**Comment by kaikreuzer** [comment]
Thanks!

File: bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/thing/ThingResource.java [hunk 4]
```diff
@@ -490,10 +490,10 @@ public Response updateConfiguration(@HeaderParam(HttpHeaders.ACCEPT_LANGUAGE) St
             logger.debug("Config description validation exception occurred for thingUID {} - Messages: {}", thingUID,
                     ex.getValidationMessages());
             return Response.status(Status.BAD_REQUEST).entity(ex.getValidationMessages(locale)).build();
-        } catch (IllegalArgumentException ex) {
-            logger.info("Received HTTP PUT request for update config at '{}' for the unknown thing '{}'.",
-                    uriInfo.getPath(), thingUID);
-            return getThingNotFoundResponse(thingUID);
+        } catch (Exception ex) {
```
Comment (maggu2810) [inline bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/thing/ThingResource.java]: I am still not a fan about catching every exception. Every unchecked one is okay for such cases but to catch also every checked one is IMHO broken.

Comment (maggu2810) [inline bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/thing/ThingResource.java]: Thinking a little bit more about the location, I assume for a rest resource it is okay... Nothing can be done in this case.

**Pull request merged**